src/lib/sql/sql_translator.cpp

```diff
@@ -619,25 +617,29 @@ SQLTranslator::TableSourceState SQLTranslator::_translate_predicated_join(const
    */
   auto lqp = std::shared_ptr<AbstractLQPNode>{};
 
-  const auto join_predicate_iter =
-      std::find_if(join_predicates.begin(), join_predicates.end(), [&](const auto& join_predicate) {
-        return is_trivial_join_predicate(*join_predicate, *left_input_lqp, *right_input_lqp);
-      });
+  if (join_mode != JoinMode::Inner && join_predicates.size() > 1) {
+    lqp = JoinNode::make(join_mode, join_predicates, left_input_lqp, right_input_lqp);
+  } else {
+    const auto join_predicate_iter =
+        std::find_if(join_predicates.begin(), join_predicates.end(), [&](const auto& join_predicate) {
+          return is_trivial_join_predicate(*join_predicate, *left_input_lqp, *right_input_lqp);
+        });
 
-  AssertInput(join_mode == JoinMode::Inner || join_predicate_iter != join_predicates.end(),
-              "Non column-to-column comparison in join predicate only supported for inner joins");
+    AssertInput(join_mode == JoinMode::Inner || join_predicate_iter != join_predicates.end(),
+                "Non column-to-column comparison in join predicate only supported for inner joins");
```

Comment:
Is this still valid? Could you add an example?

Reply:
Added 
```
    // Inner Joins with predicates like `5 + t0.a = 6+ t1.b` can be supported via Cross join + Scan. For all other join
    // modes such predicates are not supported.
```